dev-tools/reconfigurator-cli/src/main.rs

```diff
@@ -828,7 +828,20 @@ fn cmd_blueprint_edit(
                 .context("failed to add CockroachDB zone")?;
             format!("added CockroachDB zone to sled {}", sled_id)
         }
-        BlueprintEditCommands::ExpungeZone { sled_id, zone_id } => {
+        BlueprintEditCommands::ExpungeZone { zone_id } => {
+            let mut parent_sled_id = None;
+            for sled_id in builder.sled_ids_with_zones() {
+                if builder
+                    .current_sled_zones(sled_id, BlueprintZoneFilter::All)
+                    .any(|z| z.id == zone_id)
```

Comment:
Should this match all zones or some subset?

Reply:
I think `All` is right - in this case the user passed in an explicit zone ID, and we're just trying to find it. This would allow someone to try to use reconfigurator-cli to expunge an already-expunged zones, but that seems fine, I think?